admin/orders.php

```diff
@@ -1326,7 +1339,19 @@ function couponpopupWindow(url) {
                     $contents[] = array('align' => 'text-center', 'text' => '<a href="' . zen_href_link(FILENAME_ORDERS_INVOICE, 'oID=' . $oInfo->orders_id) . '" target="_blank" class="btn btn-info" role="button">' . IMAGE_ORDERS_INVOICE . '</a> <a href="' . zen_href_link(FILENAME_ORDERS_PACKINGSLIP, 'oID=' . $oInfo->orders_id) . '" target="_blank" class="btn btn-info" role="button">' . IMAGE_ORDERS_PACKINGSLIP . '</a>');
                     $zco_notifier->notify('NOTIFY_ADMIN_ORDERS_MENU_BUTTONS', $oInfo, $contents);
 
+                    // each contents array is drawn in a div, so this form block must be a single array element.
+                    $contents[] = ['text' =>
+                        zen_draw_form('statusUpdate', FILENAME_ORDERS, zen_get_all_get_params(['action','language']) . 'action=update_order' . (!isset($_GET['oID']) ? '&oID=' . $oInfo->orders_id : '') . '&language=' . $oInfo->language_code, 'post', '', true) . // form action uses the order language to change the session language on the update. On initial page load (from another page), $_GET['oID'] is not set, hence clause in form action
+                        zen_draw_hidden_field('listing', '1') . // identify that this form was submitted from infoBox/listing page and not the order details page (for the redirect to this same page)
```

Comment:
For `listing` I wonder if it makes sense to use `cameFrom`. This would retain semantic meaning, and be similar to what's already used with session timeouts.

Reply:
As it is. only the listing page provides this as post, so the redirect check is simple.
This is easily changed to cameFrom (I note all instances of this are camefrom, no camel case), but with what value?
From my vague understanding of camefrom, camefrom is a GET and should be a full filename with parameters?

Reply:
You're right: it's not camelCase.  I wondered about it, but didn't go check.

If setting the filename and parameters works from this point in the code then it may be good to remain consistent.

One thing I'm trying to avoid is the introduction of new fields that aren't already covered by Admin Sanitization. I'll post about that in a separate comment.

Reply:
ah, so its not to "use" the value, its just to allow the parameter name since camefrom is already permitted.

Reply:
Done.